marlowe-playground-client/src/Projects/Types.purs

```diff
@@ -0,0 +1,45 @@
+module Projects.Types where
+
+import Prelude
+import Analytics (class IsEvent, Event)
+import Analytics as A
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Lens (Lens')
+import Data.Lens.Record (prop)
+import Data.Maybe (Maybe(..))
+import Data.Symbol (SProxy(..))
+import Gist (Gist, GistId)
+import Network.RemoteData (RemoteData(..))
+
+data Lang
+  = Marlowe
+  | Haskell
+  | Javascript
+  | Blockly
+
+derive instance genericLang :: Generic Lang _
+
+instance showLang :: Show Lang where
+  show lang = genericShow lang
+
+data Action
+  = LoadProjects
+  | LoadProject Lang GistId
+
+defaultEvent :: String -> Event
+defaultEvent s = A.defaultEvent $ "Projects." <> s
+
+instance isEventAction :: IsEvent Action where
+  toEvent LoadProjects = Just $ defaultEvent "LoadProjects"
+  toEvent (LoadProject lang _) = Just <<< defaultEvent $ "LoadProject." <> show lang
```

Comment:
`lang` should probably become the event's `label` field. Don't mash it all into the `action`.